### PR TITLE
Implement multi rule editor for color, radius and opacity

### PR DESCRIPTION
**Pull request by hwbllmnn**
Title says it all.

@terrestris/devs Please review.

selection_006

selection_007

selection_008

### CODE REVIEW

**Comment by KaiVolland** [comment]
Screenshot plz.

File: src/Component/Style/Style.tsx [hunk 1]
```diff
@@ -246,7 +322,69 @@ export class Style extends React.Component<StyleProps, StyleState> {
...
+          wrapClassName="gs-modal-opacity"
+          footer={null}
+          onCancel={() => this.setState({opacityModalVisible: false})}
+        >
+          <OpacityField
+            opacity={opacity}
+            label={locale.opacityLabel}
+            onChange={this.updateMultiOpacities}
+          />
+        </Modal>
```
Comment (KaiVolland) [inline src/Component/Style/Style.tsx]: Please create a new Component for this.

Comment (hwbllmnn) [reply src/Component/Style/Style.tsx]: Done in 6a1fca6

File: src/Component/Symbolizer/BulkEditModals/BulkEditModals.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+}
+
+// default props
+interface BulkEditModalsDefaultProps {
+  colorModalVisible: boolean;
+  sizeModalVisible: boolean;
+  opacityModalVisible: boolean;
+  style: GsStyle;
+  locale: StyleLocale;
+  compact: boolean;
```
Comment (jansule) [inline src/Component/Symbolizer/BulkEditModals/BulkEditModals.tsx]: What's the purpose of this property? It looks like it's not being used.

File: src/Component/Style/Style.tsx [hunk 3]
```diff
@@ -175,6 +189,46 @@ export class Style extends React.Component<StyleProps, StyleState> {
...
+    const style = _cloneDeep(this.state.style);
+    const selectedRules = style.rules.filter((rule: GsRule, index: number) => {
+      return this.state.selectedRowKeys.includes(index);
+    });
+    selectedRules.forEach((rule: GsRule) => {
+      rule.symbolizers.forEach((sym: GsSymbolizer) => {
+        sym[property] = value;
+      });
+    });
+    this.setState({style});
```
Comment (jansule) [inline src/Component/Style/Style.tsx]: You should also call `onStyleChange` here, if it exists.

File: src/locale/de_DE.js [hunk 4]
```diff
@@ -6,6 +6,11 @@ export default {
         graphicalEditor: 'Graphischer Editor',
         codeEditor: 'Code Editor'
     },
+    GsBulkEditModals: {
+        colorLabel: 'Farbe wählen',
+        radiusLabel: 'Radius festlegen',
```
[the remark this answers is not in the document]
Comment (hwbllmnn) [reply src/locale/de_DE.js]: Changed in 8a6bb9b

File: src/Component/Symbolizer/BulkEditModals/BulkEditModals.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+  public static defaultProps: BulkEditModalsDefaultProps = {
+    style: {
+      name: 'My Style',
+      rules: []
+    },
+    colorModalVisible: false,
+    sizeModalVisible: false,
+    opacityModalVisible: false,
+      compact: false,
```
[the remark this answers is not in the document]
Comment (hwbllmnn) [reply src/Component/Symbolizer/BulkEditModals/BulkEditModals.tsx]: Changed in 8a6bb9b

File: src/Component/Symbolizer/BulkEditModals/BulkEditModals.tsx (hunk 2 above)
Comment (hwbllmnn) [reply src/Component/Symbolizer/BulkEditModals/BulkEditModals.tsx]: True, I removed all references in 8a6bb9b

File: src/Component/Style/Style.tsx (hunk 3 above)
Comment (hwbllmnn) [reply src/Component/Style/Style.tsx]: Changed in 8a6bb9b

**Comment by hwbllmnn** [comment]
Thanks for review! I've pushed a commit addressing your notes which also adds bulk symbol editing.

File: src/Component/Symbolizer/BulkEditModals/BulkEditModals.tsx [hunk 6]
```diff
@@ -128,6 +115,10 @@ export class BulkEditModals extends React.Component<BulkEditModalsProps, BulkEdi
         if (sym.opacity) {
           opacity = sym.opacity;
         }
+        if (sym.symbol) {
+          symbol = sym.wellKnownName;
+        }
+        console.log(sym)
```
Comment (jansule) [inline src/Component/Symbolizer/BulkEditModals/BulkEditModals.tsx]: Please remove console.log

Comment (jansule) [reply src/Component/Symbolizer/BulkEditModals/BulkEditModals.tsx]: Too fast =)

**Pull request merged**